### PR TITLE
#62 Implement volunteers list export

### PR DESCRIPTION
**Pull request by devilankur18**

### CODE REVIEW

**Comment by devilankur18** [comment]
@anandology Please review #69

File: cleansweep/volunteers/views.py [hunk 1]
```diff
@@ -33,6 +34,18 @@ def add_volunteer(place):
...
 
+@plugin.place_view("/volunteers/download")
+def download_volunteer(place):
+    headers = ['Name', "Phone", 'Email', 'Voter ID', 'Location', 'Timestamp']
+    data = tablib.Dataset(headers=headers)
+    for m in place.get_all_members():
+        data.append([m.name, m.phone, m.email, m.voterid, m.place.name, m.created.isoformat()])
+    response = make_response(data.xls)
+    response.headers['content_type'] = 'application/vnd.ms-excel;charset=utf-8'
+    response.headers['Content-Disposition'] = "attachment; filename=export.xls"
```
Comment (anandology) [inline cleansweep/volunteers/views.py]: Could you please name the file as place-key + "-volunteers.xls'. So for `DL/AC001`, it'll be `DL-AC001-volunteers.xls`.

File: cleansweep/volunteers/views.py [hunk 2]
```diff
@@ -33,6 +34,18 @@ def add_volunteer(place):
     return render_template("add_volunteer.html", place=place, form=form)
 
 
+@plugin.place_view("/volunteers/download")
+def download_volunteer(place):
+    headers = ['Name', "Phone", 'Email', 'Voter ID', 'Location', 'Timestamp']
+    data = tablib.Dataset(headers=headers)
+    for m in place.get_all_members():
+        data.append([m.name, m.phone, m.email, m.voterid, m.place.name, m.created.isoformat()])
```
Comment (anandology) [inline cleansweep/volunteers/views.py]: I don't think we need the timestamp field.

File: cleansweep/volunteers/views.py [hunk 3]
```diff
@@ -33,6 +34,18 @@ def add_volunteer(place):
     return render_template("add_volunteer.html", place=place, form=form)
 
 
+@plugin.place_view("/volunteers/download")
```
Comment (anandology) [inline cleansweep/volunteers/views.py]: I suggest making the URL as /volunteers.xls

Considering REST principles, it is not a good idea to have verbs in the URLs. Nouns are better as URLs represent resources.

**Comment by devilankur18** [comment]
@anandology updated as per suggestions.

File: cleansweep/volunteers/views.py [hunk 4]
```diff
@@ -33,6 +34,19 @@ def add_volunteer(place):
...
 
+@plugin.place_view("/volunteers.xls", permission="view-volunteers")
+def download_volunteer(place):
+    headers = ['Name', "Phone", 'Email', 'Voter ID', 'Location']
+    data = tablib.Dataset(headers=headers)
+    for m in place.get_all_members():
+        data.append([m.name, m.phone, m.email, m.voterid, m.place.name])
+    response = make_response(data.xls)
+    response.headers['content_type'] = 'application/vnd.ms-excel;charset=utf-8'
+    response.headers['Content-Disposition'] = "attachment; filename='{0}-volunteers.xls'".format(place.key)
```
Comment (anandology) [inline cleansweep/volunteers/views.py]: That would ad filename as DL/AC001-volunteers.xls. Could you please make it DL-AC001-volunteers.xls.

I think you'll have to pass `place.key.replace("/", "-")` to `format`.

Comment (devilankur18) [inline cleansweep/volunteers/views.py]: @anandology  I also thought that, but that seems to working as alright and generate the required format.

**Comment by devilankur18** [comment]
@charanjeetaap I have change the permission to write, so only power users, who have the permission to add new volunteers will only be able to download.

@anandology please review this again and merge.

**Pull request merged**